packages/medusa/src/services/product-category.ts

```diff
@@ -116,6 +119,29 @@ class ProductCategoryService extends TransactionBaseService {
     })
   }
 
+  /**
+   * Updates a product category
+   * @param productCategoryId - id of product category to update
+   * @param productCategoryInput - parameters to update in product category
+   * @return updated product category
+   */
+  async update(
+    productCategoryId: string,
+    productCategoryInput: UpdateProductCategoryInput
+  ): Promise<ProductCategory> {
+    return await this.atomicPhase_(async (manager) => {
+      const productCategoryRepo = manager.getCustomRepository(
+        this.productCategoryRepo_
+      )
+
+      const productCategory = await this.retrieve(productCategoryId)
+
+      productCategoryRepo.merge(productCategory, productCategoryInput)
```

Comment:
comment: Haven't seen `merge` before, but I kinda like it. That said, should we stick with the `for each` for consistency?

Reply:
If we go with merge, I believe we have to correct the unit tests :)

Reply:
Are you sure? merge will add both props to the new output entity, so only the props from the inputs will get updated

Reply:
reverted back to the for loop. I feel weird re-writing merge logic inside the repository mocks 😄